testsuite/features/support/file_management.rb

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2023 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+# Attempts to retrieve the SHA256 checksum for a file inside a given directory or download it
+# from the same domain the file has been downloaded from.
+# Returns the path to the checksum or nil if none has been found/downloaded
+def get_checksum_path(dir, original_file_name, file_url)
+  checksum_file_names = %W[CHECKSUM SHA256SUMS sha256sum.txt #{original_file_name}.CHECKSUM #{original_file_name}.sha256]
+
+  server = get_target('server')
+  # When using a mirror, the checksum file should be present and in the same directory of the file
+  if $mirror
+    checksum_file_names.each do |name|
+      checksum_path = "#{dir}/#{name}"
+      _output, code = server.run("test -e #{checksum_path}", check_errors: false)
+      return checksum_path if code.zero?
```

Comment:
In the way this is designed we will be doing calls to the server for each item on the checksum_file_names.
What aboout performing an unique call to the server, using an `ls` for instance, so we retrieve all the files in `dir` and after that, we can process this output to dedice if the server contains the checksum we are looking for?

Reply:
Good point, that actually makes much more sense and speeds things up a bit.